### DISCUSSION TITLE
Suggest open-source packages for a rich text editor similar to DOCX, with features such as collaboration, track changes, formatting, tables, and pagination.

### DISCUSSION THREAD

**Discussion by tkmpraveens** [ianstormtaylor/slate#5352] [Q&A]
While building a rich text editor using Slate.js, I used the slate-collaborative npm package to enable collaborative features. However, I ran into issues because there was no package available for incorporating track changes, and the table had several bugs. Additionally, there were no predefined styling options for features such as font color, font family, alignment etc.

As a result, I am searching for an open-source rich text editor that resembles DOCX and includes the following features:

- Collaboration
- Track changes
- Rich formatting options
- Tables
- Pagination

Additionally, I would prefer the packages to be customizable, stable, have a supportive community, and come with good documentation.

Although I have experimented with several open-source packages, including Quill and Draft.js, it appears that we must develop our own editor, despite their provision of fundamental features. Additionally, I have explored some paid alternatives, such as CKEditor and Syncfusion, but their cost seems unsustainable in the long term.

If anyone could recommend any open-source packages that possess all of these features, it would be highly appreciated. I have been searching for a while, but I have yet to find a suitable set of packages that satisfy all of these requirements. Thank you in advance!

**Comment by charanjit-singh** [comment]
@tkmpraveens Try https://plate.udecode.io/ for basic plugins and building your own on top of that. Demo: https://codesandbox.io/s/esgwct?file=%2FApp.tsx&from-sandpack=true

**Reply by tkmpraveens** [reply]
@charanjit-singh Thank you for your response.
